it/v1/endpoints/RetrievePeriodSummaryControllerISpec.scala

```diff
@@ -358,8 +344,8 @@ class RetrievePeriodSummaryControllerISpec extends IntegrationBaseSpec {
           ("AA123", "XAIS12345678910", "2023-04-01_2024-01-01", TaxYear.fromMtd("2023-24"), Status.BAD_REQUEST, NinoFormatError),
           ("AA123456A", "203100", "2023-04-01_2024-01-01", TaxYear.fromMtd("2023-24"), Status.BAD_REQUEST, BusinessIdFormatError),
           ("AA123456A", "XAIS12345678910", "2020", TaxYear.fromMtd("2023-24"), Status.BAD_REQUEST, PeriodIdFormatError),
-          ("AA123456A", "XAIS12345678910", "2023-04-01_2024-01-01", TaxYear.fromMtd("2021-22"), Status.BAD_REQUEST, RuleTaxYearNotSupportedError),
-          ("AA123456A", "XAIS12345678910", "2023-04-01_2024-01-01", TaxYear.fromMtd("2023-2"), Status.BAD_REQUEST, TaxYearFormatError)
+          ("AA123456A", "XAIS12345678910", "2023-04-01_2024-01-01", TaxYear.fromMtd("2023-2"), Status.BAD_REQUEST, TaxYearFormatError),
+          ("AA123456A", "XAIS12345678910", "2023-04-01_2024-01-01", TaxYear.fromMtd("2021-22"), Status.BAD_REQUEST, InvalidTaxYearParameterError)
```

Comment:
Sorry -- didn't notice this before -- but because it is using a `TaxYear` object here, it cannot accurately create the actual request url in cases where the tax year is bad. So "2023-2" will render into the request URL as "202" I think (the wrong format but not what you would expect).

It also means that you are unable to test the `RuleTaxYearRangeInvalidError` scenario.  Better to use a string for the `taxYear` in the `TysIfsTest`.

So this needs fixing and the `RuleTaxYearRangeInvalidError` scenario needs adding here and also in `ListPeriodSummariesControllerISpec`.

Reply:
Fixed